src/pages/api/waivers/index.js

```diff
@@ -46,8 +47,20 @@ const apiRoute = nextConnect({
 apiRoute.post('/api/waivers/', upload.fields([
     { name: 'adult', maxCount: 1 },
     { name: 'minor', maxCount: 1 }
-  ]), async (req, res) => {
-    res.status(200).json({ data: 'success' });
+  ]), async (req, res, next) => {
+    try {
+        const session = await getSession({ req })
+        if (session) {
+            if (session.user.role !== 'admin') {
```

Comment:
@jwu6492 I think the problem with putting the check here is that the file would be first written by multer and then the check happens here. So even though we throw an error, the user can effectively re-write the waiver file. We would need to put this check before the file write happens in `upload.fields` .